packages/jest-diff/src/diffStrings.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2014, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+const chalk = require('chalk');
+const jsDiff = require('diff');
+
+const ANNOTATION = `${chalk.red('- expected')} ${chalk.green('+ actual')}\n\n`;
+
+// diff characters if oneliner and diff lines if multiline
+function diffStrings(a: string, b: string): ?string {
+  const multiline = a.match(/[\r\n]/) !== -1 && b.indexOf('\n') !== -1;
+  let isDifferent = false;
+  let result;
+
+  if (multiline) {
+    result = jsDiff.diffLines(a, b).map(part => {
+      if (part.added || part.removed) {
+        isDifferent = true;
+      }
+
+      const lines = part.value.split('\n');
+      const color = part.added ? 'green' : part.removed ? 'red' : 'white';
+
+      if (lines[lines.length - 1] === '') {
+        lines.pop();
+      }
+
+      return lines.map(line => {
+        const mark = part.added
+          ? chalk.green('+')
+          : part.removed ? chalk.red('-') : ' ';
+        return mark + ' ' +  chalk[color](line) + '\n';
+      }).join('');
+    }).join('');
+  } else {
+    result = jsDiff.diffChars(a, b).map(part => {
+      if (part.added || part.removed) {
+        isDifferent = true;
+      }
+
+      const color = part.added ? 'green' : part.removed ? 'red' : 'gray';
+      return chalk[color](part.value);
```

Comment:
please don't use dynamic invocations like you do in this file. It's hard to statically type and hard to follow the code.

Instead I recommend:
```js
const color = part.added ? chalk.green : (part.removed ? chalk.red : chalk.gray);
```

it isn't much more complex and way easier to read (also see avove for multiline strings). I may also recommend one added set of parenthesis for readability :P